tests/metrics_publishing.rs

```diff
@@ -156,126 +131,144 @@ async fn f64_histogram_is_exported() {
     let (subscriber, exporter) = init_subscriber(
         "abcdefg_racecar".to_string(),
         InstrumentKind::Histogram,
-        NumberKind::F64,
-        Number::from(777.0012_f64),
+        777.0012_f64,
     );
 
     tracing::subscriber::with_default(subscriber, || {
-        tracing::info!(value.abcdefg_racecar = 777.0012_f64);
+        tracing::info!(histogram.abcdefg_racecar = 777.0012_f64);
     });
 
     exporter.export().unwrap();
 }
 
-fn init_subscriber(
+fn init_subscriber<T>(
     expected_metric_name: String,
     expected_instrument_kind: InstrumentKind,
-    expected_number_kind: NumberKind,
-    expected_value: Number,
-) -> (impl Subscriber + 'static, TestExporter) {
-    let controller = opentelemetry::sdk::metrics::controllers::basic(processors::factory(
-        selectors::simple::histogram(vec![-10.0, 100.0]),
-        aggregation::cumulative_temporality_selector(),
-    ))
-    .build();
+    expected_value: T,
+) -> (impl Subscriber + 'static, TestExporter<T>) {
+    let reader = ManualReader::builder()
+        .with_aggregation_selector(Box::new(DefaultAggregationSelector::new()))
+        .with_temporality_selector(DefaultTemporalitySelector::new())
+        .build();
+    let reader = TestReader {
+        inner: Arc::new(reader),
+    };
+
+    let provider = MeterProvider::builder().with_reader(reader.clone()).build();
     let exporter = TestExporter {
         expected_metric_name,
         expected_instrument_kind,
-        expected_number_kind,
         expected_value,
-        controller: controller.clone(),
+        reader,
+        _meter_provider: provider.clone(),
```

Comment:
We need hold `MeterProvider` until export complete.

`MeterProvider` hold `Pipeline` in `Arc`
https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/src/metrics/meter_provider.rs#L28

`ManualReaderInner` hold reference to `Pipeline` in `Weak`
https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/src/metrics/manual_reader.rs#L55

so, if not hold `MeterProvider`

```rust
tracing::subscriber::with_default(subscriber, || {
        tracing::info!(monotonic_counter.float_hello_world = 1.000000123_f64);
 });
// <- drop MeterProvider holded in MetricLayer

 exporter.export().unwrap(); // Weak<Pipeline>::upgrade() failed
```

Reply:
Yep this looks right